src/session/session_api.c

```diff
@@ -674,6 +671,32 @@ __session_alter(WT_SESSION *wt_session, const char *uri, const char *config)
     API_END_RET_NOTFOUND_MAP(session, ret);
 }
 
+/*
+ * __session_alter --
+ *     Alter a table setting.
+ */
+static int
+__session_alter(WT_SESSION *wt_session, const char *uri, const char *config)
+{
+    WT_DECL_RET;
+    WT_SESSION_IMPL *session;
+
+    session = (WT_SESSION_IMPL *)wt_session;
+
+    /*
+     * Alter table can return EBUSY error when the table is modified in parallel by eviction. Retry
+     * the command after performing a system wide checkpoint. Don't retry it in a loop as it may
+     * lead to wait forever.
+     */
+    ret = __session_alter_internal(session, uri, config);
+    if (ret == EBUSY) {
+        WT_RET(session->iface.checkpoint(&session->iface, NULL));
```

Comment:
I am assuming it is still possible that eviction activity will mark the tree dirty between calling checkpoint and alter?

Reply:
Yes, it is possible the tree can be marked dirty by eviction. The problem here is there is no guarantee that after a fixed number of checkpoints, the tree can never be dirty.  I went with the implementation of the best scenario where it doesn't happen.